tests/php/View/Parsers/HTMLValueTest.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+namespace SilverStripe\View\Tests\Parsers;
+
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\View\Parsers\HTMLValue;
+use SilverStripe\ORM\FieldType\DBHTMLText;
+use SilverStripe\View\Parsers\ShortcodeParser;
+use SilverStripe\Core\Convert;
+
+class HTMLValueTest extends SapphireTest
+{
+    public function testInvalidHTMLParsing()
+    {
+        $value = new HTMLValue();
+
+        $invalid = [
+            '<p>Enclosed Value</p><p>a' => '<p>Enclosed Value</p><p>a</p>',
```

Comment:
There is a very minor difference in this test vs https://github.com/silverstripe/silverstripe-html5/blob/2/tests/HTML5ValueTest.php#L22 where we use `<p>a` here vs `</p>` in the other test - reason being that using masterminds/html5 the `</p>` will simply be removed, which is a better behavior IMO